install/kubernetes/cilium/values.yaml

```diff
@@ -796,6 +796,12 @@ hubble:
             matchLabels:
               k8s-app: cilium
 
+    # -- Pod topology spread constraints for hubble-relay
```

Comment:
After #20066 was merged, now these changes need to be made to `install/kubernetes/cilium/values.yaml.tmpl` instead, then run `make -C install/kubernetes` to update this file.

Reply:
Hmm tried to revert the changes inside `values.yaml` and only implement them inside the `values.yaml.tmpl`. Make will not update the real yaml, is that correct?

Reply:
:thinking: `install/kubernetes/cilium/values.yaml` is not a dependency for itself, so that's why make doesn't rebuild it if you change it. However if you change `install/kubernetes/cilium/values.yaml.tmpl` then it should regenerate.

I'm not sure it makes sense to me for it to depend on changes to itself, that seems like it will likely only argue with developers that attempt to change it. Is there precedence for making make targets for files dependent on the file that's generated?

Reply:
I can do some testing on this but I am not familiar with complex Makefiles like you use in this project 😆

Reply:
Personally I wouldn't worry about it, I don't think it makes sense. Now that the `values.yaml` starts with the following lines, then future contributors should find out pretty quickly that they're editing the wrong file if they attempt to change this one:

```
# File generated by install/kubernetes/Makefile; DO NOT EDIT.                                                                                                                                                
# This file is based on install/kubernetes/cilium/values.yaml.tmpl.